### DISCUSSION TITLE
Random color priority for plots

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#2657] [Ideas]
Lay your eyes upon this plot:

![image](https://github.com/user-attachments/assets/958fecfb-b06e-41b5-a351-e09f8a7314ed)

There are two columns, but visidata is only showing two of them. What's happening here is that the three columns have similar data, so they hide each other. It appears that VisiData prioritizes showing column 2 over column 1 over column 0. This means that we can visually discern that columns 1 and 2 have roughly the same data, but we have to guess what's happening with column 0.

If VisiData would have had a different, random priority of columns for each point in the plot, then we could see that the three columns are the same.

What do you think?

**Comment by saulpw** [comment]
options.disp_pixel_random is what you want (and it should be renamed to include `graph`).

**Reply by cool-RR** [reply]
Awesome, thank you! It's funny how it sweeps through the colors for a few seconds.

**Reply by cool-RR** [reply]
I'm a little confused by how it twinkles through colors. Sometimes it does it for just a few seconds, sometimes for a way longer period. For a few seconds it's cute, but more than that is distracting. I wonder what determines how long it'll twinkle.

**Reply by saulpw** [reply]
Movement/keypresses.  It settles down into idle mode after a second but then un-idles if you do anything.
